### ISSUE TITLE
support for tox into setup.cfg

### ISSUE THREAD

**Issue by gpongelli** [mgedmin/check-python-versions#35]
Tox can be configured also into setup.cfg file, e.g.

```
[tox:tox]
isolated_build = true
envlist = lint, format, py3.8, py3.9, py3.10, py3.11, build
```

or, using default naming convention
```
[tox:tox]
isolated_build = true
envlist = lint, format, py38, py39, py310, py311, build
```

**Comment by mgedmin** [comment]
Wait.  tox allows dots in environment names?  tox -e py3.11?  Whoa.  That should probably be a separate issue.

**Comment by jugmac00** [comment]
This is a surprise for me, too. I will check with the other maintainers whether this was introduced intentionally with tox v4 or not.

**Comment by jugmac00** [comment]
Ok, looks like this is a feature, and not even a new one, see https://tox.wiki/en/latest/user_guide.html#test-environments

**Comment by gpongelli** [comment]
> Wait. tox allows dots in environment names? tox -e py3.11? Whoa. That should probably be a separate issue.

I don't know. I use this dotted notation because, having poetry that run tox environments locally and remotely, I've encountered errors when using normal notation (py39 , py310...) .
This is because I need to use poetry environments (where packages from poetry.lock are installed, also tox) instead of directly running tox environments (if so, there is a duplication of package lists to be installed: first list into pyproject.toml for poetry env and then same list into setup.cfg/tox.ini for tox); I obviously prefer having only one source of truth instead of maintaining same things in different places. If you want elaborate on, [this is the cookiecutter template](https://github.com/gpongelli/cookiecutter-pypackage/tree/gpongelli-patch-1) I've created and constantly use to create new packages using both poetry and tox.

Reading the [link](https://tox.wiki/en/latest/user_guide.html#test-environments) from @jugmac00 , seems it's "supported" : `You can also specify these factors with a period between the major and minor versions (e.g. pyN.M)` , but I use them just for [poetry environment switching](https://github.com/gpongelli/cookiecutter-pypackage/blob/gpongelli-patch-1/%7B%7Bcookiecutter.project_slug%7D%7D/setup.cfg#L100-L103).

**Comment by gpongelli** [comment]
I've done changes on local branch for this issue, but it depends on PR #39 so I wait to push it until dependent PR will be merged 😉
